src/hooks/fix-decompress.ts

```diff
@@ -0,0 +1,93 @@
+import zlib from 'zlib';
+import { ClientRequest, IncomingMessage } from 'http';
+import { HandlerFunction } from 'got-cjs';
+import { PassThrough, Transform } from 'stream';
+import mimicResponse from 'mimic-response';
+
+const onResponse = (response: IncomingMessage, propagate: (fixedResponse: IncomingMessage) => void) => {
+    const encoding = response.headers['content-encoding']?.toLowerCase();
+
+    // Append empty chunk.
+    const zlibOptions = {
+        flush: zlib.constants.Z_SYNC_FLUSH,
+        finishFlush: zlib.constants.Z_SYNC_FLUSH,
+    };
+
+    const useDecompressor = (decompressor: Transform) => {
+        delete response.headers['content-encoding'];
+
+        const result = new PassThrough({
+            autoDestroy: false,
+            destroy(error, callback) {
+                response.destroy();
+
+                callback(error);
+            },
+        });
+
+        response.pipe(decompressor).pipe(result);
+
+        propagate(mimicResponse(response, result));
+    };
+
+    if (encoding === 'gzip' || encoding === 'x-gzip') {
+        useDecompressor(zlib.createGunzip(zlibOptions));
+    } else if (encoding === 'deflate' || encoding === 'x-deflate') {
+        let read = false;
+
+        response.once('data', (chunk: Buffer) => {
+            read = true;
+
+            response.unshift(chunk);
+
+            // See http://stackoverflow.com/questions/37519828
+            // eslint-disable-next-line no-bitwise
+            const decompressor = (chunk[0] & 0x0F) === 0x08 ? zlib.createInflate() : zlib.createInflateRaw();
+            useDecompressor(decompressor);
+        });
+
+        response.once('end', () => {
+            if (!read) {
+                propagate(response);
+            }
+        });
+    } else if (encoding === 'br') {
+        useDecompressor(zlib.createBrotliDecompress());
+    } else {
+        propagate(response);
+    }
+};
+
+export const fixDecompress: HandlerFunction = (options, next) => {
+    if (options.decompress) {
+        options.headers['accept-encoding'] = process.versions.brotli ? 'gzip, deflate, br' : 'gzip, deflate';
```

Comment:
Isn't brotli always supported in Node.js 15+?

Also, we probably don't need to add the header, because we use the auto-generated headers that match the browser. I assume those headers include accept encoding as well, but I have not checked TBH.

Reply:
https://github.com/apify/got-scraping/runs/4689723092?check_suite_focus=true#step:5:13 failed, so I assumed this was running Node.js 14, but indeed it was 15.

Nice catch!